server/src/test/java/org/elasticsearch/index/mapper/IpPrefixAutomatonUtilTests.java

```diff
@@ -76,94 +77,105 @@ public void testIncompleteDecimalGroupAutomaton() throws UnknownHostException {
 
         }
         {
-            CompiledAutomaton compiledAutomaton = IpPrefixAutomatonUtil.buildIpPrefixAutomaton("1.");
-            byte[] encode = InetAddressPoint.encode(InetAddress.getByName("1.2.3.4"));
-            assertTrue(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("10.2.3.4"));
-            assertFalse(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("2.2.3.4"));
-            assertFalse(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
+            CompiledAutomaton a = buildIpPrefixAutomaton("0");
+            assertTrue(accepts(a, "0.2.3.4"));
+            assertFalse(accepts(a, "::1.1.2.3"));
+            assertFalse(accepts(a, "1c7:21d6:ffff:ffff:6852:f279::"));
         }
         {
-            CompiledAutomaton compiledAutomaton = IpPrefixAutomatonUtil.buildIpPrefixAutomaton("1:2");
-            byte[] encode = InetAddressPoint.encode(InetAddress.getByName("1:2::1"));
-            assertTrue(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("1:2a::1"));
-            assertTrue(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("1:2ab::1"));
-            assertTrue(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("1:2ab5::1"));
-            assertTrue(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("10:2::3:4"));
-            assertFalse(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("::1:2:3:4"));
-            assertFalse(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
+            CompiledAutomaton a = buildIpPrefixAutomaton("00");
+            assertFalse(accepts(a, "0.2.3.4"));
+            assertFalse(accepts(a, "::1.1.2.3"));
+            assertFalse(accepts(a, "c7:21d6:ffff:ffff:6852:f279::"));
         }
         {
-            CompiledAutomaton compiledAutomaton = IpPrefixAutomatonUtil.buildIpPrefixAutomaton("::1:2");
-            byte[] encode = InetAddressPoint.encode(InetAddress.getByName("::1:2"));
-            assertTrue(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("0:0:1:2::1"));
-            assertTrue(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("1:2ab::1"));
-            assertFalse(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
+            CompiledAutomaton a = buildIpPrefixAutomaton("0:");
```

Comment:
With an additional colon it is clear that this is ipv6, so we can match all that gets shortened to "0:"